### PR TITLE
Increase targetSdk to 29

### PR DESCRIPTION
**Pull request by onurays**

### CODE REVIEW

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/network/NetworkInfoReceiver.kt [hunk 1]
```diff
@@ -14,6 +14,8 @@
  * limitations under the License.
  */
 
+@file:Suppress("DEPRECATION")
```
Comment (BillCarsonFr) [inline matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/network/NetworkInfoReceiver.kt]: add a comment explaining why we ignore warning

File: vector/src/main/java/im/vector/riotx/features/crypto/keysbackup/setup/KeysBackupSetupActivity.kt [hunk 2]
```diff
@@ -153,7 +153,7 @@ class KeysBackupSetupActivity : SimpleFragmentActivity() {
                                         }
 
                                         override fun onFailure(failure: Throwable) {
-                                            toast(failure.localizedMessage)
+                                            failure.localizedMessage?.let { toast(it) }
```
Comment (BillCarsonFr) [inline vector/src/main/java/im/vector/riotx/features/crypto/keysbackup/setup/KeysBackupSetupActivity.kt]: Maybe use a generic error string and toast it

File: vector/src/main/java/im/vector/riotx/features/crypto/verification/VerificationBottomSheetViewModel.kt [hunk 3]
```diff
@@ -362,7 +362,9 @@ class VerificationBottomSheetViewModel @AssistedInject constructor(
                         }
                     }
                 } catch (failure: Throwable) {
-                    _viewEvents.post(VerificationBottomSheetViewEvents.ModalError(failure.localizedMessage))
+                    failure.localizedMessage?.let {
```
Comment (BillCarsonFr) [inline vector/src/main/java/im/vector/riotx/features/crypto/verification/VerificationBottomSheetViewModel.kt]: same here, use a default error string and display it (R.string.unexpected_error)

File: vector/src/main/java/im/vector/riotx/features/home/room/detail/RoomDetailFragment.kt [hunk 4]
```diff
@@ -782,7 +782,7 @@ class RoomDetailFragment @Inject constructor(
                 updateComposerText("")
             }
             is RoomDetailViewEvents.SlashCommandResultError    -> {
-                displayCommandError(sendMessageResult.throwable.localizedMessage)
+                sendMessageResult.throwable.localizedMessage?.let { displayCommandError(it) }
```
Comment (BillCarsonFr) [inline vector/src/main/java/im/vector/riotx/features/home/room/detail/RoomDetailFragment.kt]: same remark

File: vector/src/main/java/im/vector/riotx/features/settings/VectorLocale.kt [hunk 5]
```diff
@@ -59,9 +59,9 @@ object VectorLocale {
         val preferences = PreferenceManager.getDefaultSharedPreferences(context)
 
         if (preferences.contains(APPLICATION_LOCALE_LANGUAGE_KEY)) {
-            applicationLocale = Locale(preferences.getString(APPLICATION_LOCALE_LANGUAGE_KEY, ""),
-                    preferences.getString(APPLICATION_LOCALE_COUNTRY_KEY, ""),
-                    preferences.getString(APPLICATION_LOCALE_VARIANT_KEY, "")
+            applicationLocale = Locale(preferences.getString(APPLICATION_LOCALE_LANGUAGE_KEY, "")!!,
```
Comment (BillCarsonFr) [inline vector/src/main/java/im/vector/riotx/features/settings/VectorLocale.kt]: the !! is needed? even with a default value?

Comment (onurays) [reply vector/src/main/java/im/vector/riotx/features/settings/VectorLocale.kt:62]: Yes it is, unfortunately :(

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/network/NetworkInfoReceiver.kt (hunk 1 above)
Comment (onurays) [reply matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/network/NetworkInfoReceiver.kt:18]: I added the comment, but it is not marked as "outdated". Please see "Files Changed" section.

**Pull request merged**